mobile-widgets/qmlmanager.cpp

```diff
@@ -2125,3 +2128,70 @@ void QMLManager::appInitialized()
 	checkPendingIntents();
 #endif
 }
+
+void QMLManager::exportToFile(export_types type, QString dir, bool anonymize)
+{
+	// add standard filename
+	QString fileName = dir.right(dir.size() - 7) + "/Subsurface_export";
+
+	switch (type)
+	{
+		case EX_DIVES_XML:
+			save_dives_logic(qPrintable(fileName + ".ssrf"), false, anonymize);
+			break;
+		case EX_DIVE_SITES_XML:
+			{
+				std::vector<const dive_site *> sites = exportFuncs::instance()->getDiveSitesToExport(false);
+				save_dive_sites_logic(qPrintable(fileName + ".xml"), &sites[0], (int)sites.size(), anonymize);
+				break;
+			}
+		case EX_UDDF:
+			exportFuncs::instance()->exportUsingStyleSheet(fileName + ".uddf", true, 0, "uddf-export.xslt", anonymize);
+			break;
+		case EX_CSV_DIVE_PROFILE:
+			exportFuncs::instance()->exportUsingStyleSheet(fileName + ".uddf", true, 0, "xml2csv.xslt", anonymize);
+			break;
+		case EX_CSV_DETAILS:
+			exportFuncs::instance()->exportUsingStyleSheet(fileName + ".uddf", true, 0, "xml2manualcsv.xslt", anonymize);
+			break;
+		case EX_CSV_PROFILE:
+			save_profiledata(qPrintable(fileName + ".csv"), true);
+			break;
+		case EX_PROFILE_PNG:
+			exportFuncs::instance()->exportProfile(qPrintable(fileName + ".png"), false);
+			break;
+		case EX_WORLD_MAP:
+			export_worldmap_HTML(qPrintable(fileName + ".html"), true);
+			break;
+		case EX_TEX:
+			exportFuncs::instance()->export_TeX(qPrintable(fileName + ".tex"), true, true);
+			break;
+		case EX_LATEX:
+			exportFuncs::instance()->export_TeX(qPrintable(fileName + ".tex"), true, false);
+			break;
+		case EX_IMAGE_DEPTHS:
+			exportFuncs::instance()->export_depths(qPrintable(fileName), false);
+			break;
+		default:
+			break;
+	}
+}
+
+void exportFuncs::saveProfile(const struct dive *dive, const QString filename)
+{
+	// TBD
+}
+
+void QMLManager::exportToWEB(export_types type, QString userId, QString password, bool anonymize)
+{
+	switch (type)
+	{
+		case EX_DIVELOGS_DE:
+//			DivelogsDeWebServices::instance()->prepareDivesForUpload(ui->exportSelected->isChecked());
+		case EX_DIVESHARE:
```

Comment:
just reading this code is an invitation to forget adding the `break;` at the end of each case statement.
Please don't do that.

Reply:
corrected